src/core/modules/Planet.cpp

```diff
@@ -731,6 +743,7 @@ QString Planet::getInfoString(const StelCore* core, const InfoStringGroup& flags
 				oss << QString("%1: %2%").arg(q_("Eclipse obscuration")).arg(QString::number(eclipseObscuration, 'f', 2)) << "<br />";
 				if (onEarth)
 				{
+					// FIXME: This is for Solar eclipses only. Make sure it works for Mercury/Venus transits, or omit it then!
```

Comment:
Why FIXME here? This code made exactly for Solar eclipses on Earth only. Or you want to compute "Eclipse magnitude" for transits?

Reply:
Look at the weird output during a Venus transit (2004, 2012) or Mercury transit (11.11.2019). Better avoid the line if the Moon is not involved. (just check for Moon distance?)

Reply:
I see no weird data (I see not eclipse data!) in version 0.19.3. So, weird data is result of refactoring?

Reply:
Oops... I was wrong!

Reply:
OK, I have an idea how to resolve this issue

Reply:
OK, I've fixed it in master